lib/Deserializer/functions.php

```diff
@@ -346,6 +346,13 @@ function functionCaller(Reader $reader, callable $func, string $namespace)
     }
 
     $funcArgs = [];
+    /**
+     * Even if $func gets "exploded" into an array, that array is still "callable"
+     * It should have elements that are the class and the function/method in the class.
+     * The declaration here helps phpstan to understand.
+     *
+     * @var (array<int,string>&callable():mixed)|\Closure|callable-string $func
```

Comment:
without telling phpstan all about $func it thinks that the array returned by `explode()` is "any old array".
And phpstan also doesn't think that `ReflectionFunction` is being called with a valid thing.

Reply:
I added a bunch of checks in the latest commit, and adjusted the PHPdoc var declaration to just say that it is a callable.